File: app/src/main/java/com/hoymm/root/tictactoe2/DisplayMenu/MainMenu/Settings/DifficulyFragment.java
package com.hoymm.root.tictactoe2.DisplayMenu.MainMenu.Settings;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.hoymm.root.tictactoe2.R;

/**
 * Created by Damian Muca - Kaizen (27.09.17)
 */

public class DifficulyFragment extends Fragment {
    private TextView easy, medium, hard;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.settings_fragment_difficulty, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        restoreLastConfigurationAndSetOnclickBehavior();
    }


    void restoreLastConfigurationAndSetOnclickBehavior() {
        initDifficulyTextViews();
        difficultyTextsSetOnClickBehavior();
        restoreDifficultyLastConfiguration();
    }

    private void initDifficulyTextViews() {
        easy = getActivity().findViewById(R.id.easyTextID);
        medium = getActivity().findViewById(R.id.mediumTextID);
        hard = getActivity().findViewById(R.id.hardTextID);
    }

    private void difficultyTextsSetOnClickBehavior() {
        easy.setOnClickListener(getClickListenerForDifficultyTextClicked());
        medium.setOnClickListener(getClickListenerForDifficultyTextClicked());
        hard.setOnClickListener(getClickListenerForDifficultyTextClicked());
    }


    @NonNull
    private View.OnClickListener getClickListenerForDifficultyTextClicked() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                SettingsSharedPreferences.saveDifficultyId(getActivity(), view.getId());
                disableAllOfTextViews();
                view.setActivated(true);
            }
        };
    }

    private void restoreDifficultyLastConfiguration() {
        disableAllOfTextViews();
        readSharedPreferencesAndEnableLast();
    }

    private void disableAllOfTextViews() {
        easy.setActivated(false);
        medium.setActivated(false);
        hard.setActivated(false);
    }

    private void readSharedPreferencesAndEnableLast() {
        int idOfDifficulyEnabled = SettingsSharedPreferences.getDifficultyId(getActivity());
        switch (DifficultyEnum.getDifficultyFromID(idOfDifficulyEnabled)){
            case easy:
                easy.setActivated(true);
                break;
            default:
            case medium:
                medium.setActivated(true);
                break;
            case hard:
                hard.setActivated(true);
                break;
        }
    }
}

File: app/src/main/java/com/hoymm/root/tictactoe2/DisplayMenu/MainMenu/Settings/DifficulyFragment.kt
package com.hoymm.root.tictactoe2.DisplayMenu.MainMenu.Settings

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.hoymm.root.tictactoe2.R

/**
 * Created by Damian Muca - Kaizen (27.09.17)
 */

class DifficulyFragment : Fragment() {
    private var easy: TextView? = null
    private var medium: TextView? = null
    private var hard: TextView? = null


    private val clickListenerForDifficultyTextClicked: View.OnClickListener
        get() = View.OnClickListener { view ->
            SettingsSharedPreferences.saveDifficultyId(activity, view.id)
            disableAllOfTextViews()
            view.isActivated = true
        }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View?
            = inflater!!.inflate(R.layout.settings_fragment_difficulty, container, false)

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        restoreLastConfigurationAndSetOnclickBehavior()
    }


    internal fun restoreLastConfigurationAndSetOnclickBehavior() {
        initDifficulyTextViews()
        difficultyTextsSetOnClickBehavior()
        restoreDifficultyLastConfiguration()
    }

    private fun initDifficulyTextViews() {
        easy = activity.findViewById(R.id.easyTextID)
        medium = activity.findViewById(R.id.mediumTextID)
        hard = activity.findViewById(R.id.hardTextID)
    }

    private fun difficultyTextsSetOnClickBehavior() {
        easy!!.setOnClickListener(clickListenerForDifficultyTextClicked)
        medium!!.setOnClickListener(clickListenerForDifficultyTextClicked)
        hard!!.setOnClickListener(clickListenerForDifficultyTextClicked)
    }

    private fun restoreDifficultyLastConfiguration() {
        disableAllOfTextViews()
        readSharedPreferencesAndEnableLast()
    }

    private fun disableAllOfTextViews() {
        easy!!.isActivated = false
        medium!!.isActivated = false
        hard!!.isActivated = false
    }

    private fun readSharedPreferencesAndEnableLast() {
        val idOfDifficulyEnabled = SettingsSharedPreferences.getDifficultyId(activity)
        when (DifficultyEnum.getDifficultyFromID(idOfDifficulyEnabled)) {
            DifficultyEnum.easy -> easy!!.isActivated = true
            DifficultyEnum.medium -> medium!!.isActivated = true
            DifficultyEnum.hard -> hard!!.isActivated = true
        }
    }
}
